File: FluentFTP.Examples/FileExists.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {
	internal static class FileExistsExample {

		public static void FileExists() {
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				conn.Connect();

				// The last parameter forces FluentFTP to use LIST -a 
				// for getting a list of objects in the parent directory.
				if (conn.FileExists("/full/or/relative/path")) {
					// dome something
				}
			}
		}

		public static async Task FileExistsAsync() {
			var token = new CancellationToken();
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await conn.ConnectAsync(token);

				// The last parameter forces FluentFTP to use LIST -a 
				// for getting a list of objects in the parent directory.
				if (await conn.FileExistsAsync("/full/or/relative/path", token)) {
					// dome something
				}
			}
		}

	}
}

File: FluentFTP.VBExamples/FileExists.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module FileExistsExample
		Sub FileExists()
			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				conn.Connect()

				If conn.FileExists("/full/or/relative/path") Then
					' do something
				End If

			End Using
		End Sub

		Async Function FileExistsAsync() As Task
			Dim token = New CancellationToken()

			Using conn = New AsyncFtpClient("127.0.0.1", "ftptest", "ftptest")
				Await conn.Connect(token)

				If Await conn.FileExists("/full/or/relative/path", token) Then
					' do something
				End If

			End Using
		End Function
	End Module
End Namespace
